src/main/java/tutoraid/logic/commands/EditStudentCommand.java

```diff
@@ -0,0 +1,208 @@
+import tutoraid.model.student.Student;
+import tutoraid.model.student.StudentName;
+
+/**
+ * Edits the details of an existing person in the address book.
+ */
+public class EditStudentCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the student identified "
+            + "by the index number used in the displayed student list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_STUDENT_NAME + "STUDENT NAME "
+            + PREFIX_STUDENT_PHONE + "STUDENT PHONE "
+            + PREFIX_PARENT_NAME + "PARENT NAME "
+            + PREFIX_PARENT_PHONE + "PARENT PHONE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_STUDENT_NAME + "John Doe "
+            + PREFIX_STUDENT_PHONE + "81234567 "
+            + PREFIX_PARENT_NAME + "Mrs Doe "
+            + PREFIX_PARENT_PHONE + "91234567 ";
+
+    public static final String MESSAGE_EDIT_STUDENT_SUCCESS = "Edited Person: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_STUDENT = "This student already exists in TutorAid";
+
+    private final Index targetIndex;
+    private final EditStudentDescriptor editStudentDescriptor;
+
+    /**
+     * @param targetIndex           of the person in the filtered person list to edit
+     * @param editStudentDescriptor details to edit the person with
+     */
+    public EditStudentCommand(Index targetIndex, EditStudentDescriptor editStudentDescriptor) {
+        requireNonNull(targetIndex);
+        requireNonNull(editStudentDescriptor);
+
+        this.targetIndex = targetIndex;
+        this.editStudentDescriptor = new EditStudentDescriptor(editStudentDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Student> lastShownList = model.getFilteredStudentList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        Student studentToEdit = lastShownList.get(targetIndex.getZeroBased());
+        Student editedStudent = createEditedPerson(studentToEdit, editStudentDescriptor);
+
+        if (!studentToEdit.isSameStudent(editedStudent) && model.hasStudent(editedStudent)) {
+            throw new CommandException(MESSAGE_DUPLICATE_STUDENT);
+        }
+
+        model.setStudent(studentToEdit, editedStudent);
+        model.updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_STUDENT_SUCCESS, editedStudent));
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code studentToEdit}
```

Comment:
Student should be written here instead of Person
```suggestion
     * Creates and returns a {@code Student} with the details of {@code studentToEdit}
```